evm/utils/test_builder.py

```diff
@@ -0,0 +1,309 @@
+from collections import OrderedDict
+from collections.abc import (
+    Iterable,
+    Mapping,
+)
+import copy
+import itertools
+import subprocess
+
+from cytoolz import (
+    assoc_in,
+    merge,
+)
+from eth_utils import (
+    decode_hex,
+    encode_hex,
+    force_text,
+    int_to_big_endian,
+    to_checksum_address,
+    apply_formatter_to_array,
+    combine_argument_formatters,
+    curried,
+)
+from evm.utils.fixture_tests import (
+    hash_log_entries,
+)
+
+from evm.db.state import (
+    MainAccountStateDB,
+)
+from evm.db.backends.memory import MemoryDB
+
+
+DEFAULT_ENV = OrderedDict([
+    ("currentCoinbase", decode_hex("0x2adc25665018aa1fe0e6bc666dac8fc2697ff9ba")),
+    ("currentDifficulty", 131072),
+    ("currentGasLimit", 1000000),
+    ("currentNumber", 1),
+    ("currentTimestamp", 1000),
+    ("previousHash", decode_hex(
+        "0x5e20a0453cecd065ea59c37ac63e079ee08998b6045136a8ce6635c7912ec0b6"
+    )),
+])
+
+ALL_FORKS = [
+    "Frontier",
+    "Homestead",
+    "EIP150",
+    "EIP158",
+    "Byzantium"
+]
+
+FILLED_WITH_TEMPLATE = "py-evm-{version}"
+
+
+def normalize_state(state):
+    if isinstance(state, Iterable) and not isinstance(state, Mapping):
+        state = state_list_to_dict(state)
+    for address, account_state_dict in state.items():
+        account_state_dict.setdefault("balance", 0)
+        account_state_dict.setdefault("nonce", 0)
+        account_state_dict.setdefault("code", b"")
+        account_state_dict.setdefault("storage", OrderedDict())
```

Comment:
This will mutate the original state dictionary which is very likely to cause un-expected errors.  Can you change this to a non-mutating pattern?